docs/select/README.md

```diff
@@ -0,0 +1,88 @@
+# Select API Quickstart Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+Traditional retrieval of objects is always as whole entities, i.e GetObject for a 5 GiB object, will always return 5 GiB of data. S3 Select API allows us to retrieve a subset of data by using simple SQL expressions. By using Select API to retrieve only the data needed by the application, drastic performance improvements can be achieved.
+
+> This implementation is compatible with AWS S3 implementation of [Select API](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectSELECTContent.html)
+
+## 1. Prerequisites
+- Install Minio Server from [here](http://docs.minio.io/docs/minio-quickstart-guide).
+- Familiarity with [AWS S3 API](https://docs.aws.amazon.com/AmazonS3/latest/API/Welcome.html).
+- Familiarity with Python and installing dependencies.
+
+## 2. Install boto3
+Install `aws-sdk-python` from AWS SDK for Python official docs [here](https://aws.amazon.com/sdk-for-python/)
+
+## 3. Example
+As an example, let us take a gzip compressed CSV file. Without S3 Select, we would need to download, decompress and process the entire CSV to get the data you needed. With Select API, can use a simple SQL expression to return only the data from the CSV you’re interested in, instead of retrieving the entire object. Following Python example shows how to retrieve the first column `Location` from an object containing data in CSV format.
+
+Please replace ``endpoint_url``,``aws_access_key_id``, ``aws_secret_access_key``, ``Bucket`` and ``Key`` with your local setup in this ``select.py`` file.
+
+```py
+#!/usr/bin/env/env python3
+import boto3
+
+s3 = boto3.client('s3',
+                  endpoint_url='http://localhost:9000',
+                  aws_access_key_id='minio',
+                  aws_secret_access_key='minio123',
+                  region_name='us-east-1')
+
+r = s3.select_object_content(
+    Bucket='mycsvbucket',
+    Key='sampledata/TotalPopulation.csv.gz',
+    ExpressionType='SQL',
+    Expression="select * from s3object s where s.Location like '%United States%'",
+    InputSerialization={
+        'CSV': {
+            "FileHeaderInfo": "USE",
+        },
+        'CompressionType': 'GZIP',
+    },
+    OutputSerialization={'CSV': {}},
+)
+
+for event in r['Payload']:
+    if 'Records' in event:
+        records = event['Records']['Payload'].decode('utf-8')
+        print(records)
+    elif 'Stats' in event:
+        statsDetails = event['Stats']['Details']
+        print("Stats details bytesScanned: ")
+        print(statsDetails['BytesScanned'])
+        print("Stats details bytesProcessed: ")
+        print(statsDetails['BytesProcessed'])
+```
+
+## 4. Run the Program
+Upload first a sample dataset downloaded from [TotalPopulation.csv](https://esa.un.org/unpd/wpp/DVD/Files/1_Indicators%20(Standard)/CSV_FILES/WPP2017_TotalPopulationBySex.csv) using the following commands.
```

Comment:
File name downloaded is WPP2017_TotalPopulationBySex.csv, should either change file name or change the script to reference the actual file name